### PR TITLE
[JENKINS-68696] Remove `javax.servlet:servlet-api` dependency

### PR DESCRIPTION
**Pull request by basil**
See [JENKINS-68696](https://issues.jenkins.io/browse/JENKINS-68696).

https://github.com/jenkinsci/jenkins/blob/a4f9f5b64b441172f6a1d871a7c615c03ac26235/pom.xml#L115-L132= is a trick added in commit [982a8fa](https://github.com/jenkinsci/jenkins/commit/982a8faf5a853129039ba2babe2a8cb35d9aca1f) apparently to prevent plugins (and core?) from bundling Servlet API 2.x. `javax.servlet:servlet-api` has been banned through Enforcer in the plugin parent POM since jenkinsci/plugin-pom@a608967 and in the core parent POM since jenkinsci/pom@82f73fef, so it is longer necessary to keep this trick in core. Keeping it complicates `jenkinsci/jenkins/pom.xml` and our Dependabot configuration, so I think it is best to dispense with it from both the core dependency tree and the Dependendabot exclusions list and rely on the abovementioned Enforcer bans to keep it out of our dependency trees.

### Testing done

- Core test suite passes
- BOM test passed in https://github.com/jenkinsci/bom/pull/1626

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7478"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
